viewflow/lock.py

```diff
@@ -70,10 +88,36 @@ def lock(flow_cls, process_pk):
                 time.sleep(sleep_time)
         else:
             raise FlowLockFailed('Lock failed for {}'.format(flow_cls))
+        try:
+            with transaction.atomic():
+                yield
+        finally:
+            cache.delete(key)
 
-        with transaction.atomic():
-            yield
+    return lock
+
+
+def redis_lock(flow, attempts=5, expires=120):
+    """Task lock based redis locks implemented in ``django-redis``."""
```

Comment:
Since this is depends from non-core django, this should be in the 3d party module.